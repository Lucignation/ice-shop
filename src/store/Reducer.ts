import { createSlice } from '@reduxjs/toolkit';
import type { PayloadAction } from '@reduxjs/toolkit';
import { IUser, IAccount } from '../common/interfaces';
import { IProduct } from '../common/product.interface';

const initialState: IAccount = {
  user: {
    id: 0,
    email: '',
    username: '',
    password: '',
    name: {
      firstname: '',
      lastname: '',
    },
    address: {
      city: '',
      street: '',
      number: 0,
      zipcode: '',
      geolocation: {
        lat: '',
        long: '',
      },
    },
    phone: '',
  },
  token: '',
  products: [],
  product: {
    id: 0,
    title: '',
    price: '',
    category: '',
    description: '',
    image: '',
  },
  cart: [],
};

export const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    setUser: (state: IAccount, action: PayloadAction<IAccount>) => {
      //   localStorage.setItem('token', action.payload.token);
      //   // localStorage.removeItem('token');
      //   state.user = action.payload.msg;
    },
    setLogin: (state: IAccount, action: PayloadAction<IAccount>) => {
      localStorage.setItem('token', action.payload.token);
      state.token = action.payload.token;
    },
    logOut: () => {
      localStorage.removeItem('token');
    },
  },
});

// Action creators are generated for each case reducer function
export const { setUser, logOut, setLogin } = userSlice.actions;

export default userSlice.reducer;
