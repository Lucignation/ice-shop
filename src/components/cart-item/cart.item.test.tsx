import CartItem from './cart-item.component';
import renderer from 'react-test-renderer';
import Cart from '../../pages/cart/cart.page';
import { Provider } from 'react-redux';
import { store } from '../../store/store';

const product = {
  id: 1,
  title: 'product title',
  price: '10',
  category: 'product category',
  image: 'product image url',
  description: 'product description',
};

const mockProps = {
  index: 0,
  product: product,
};

describe('Cart Item ', () => {
  it('renders a snapshot of <CartItem /> when it is passed into a component ', () => {
    const cartItem = renderer
      .create(
        <Provider store={store}>
          <CartItem {...mockProps} />
        </Provider>
      )
      .toJSON();

    expect(cartItem).toMatchSnapshot();
  });
});
