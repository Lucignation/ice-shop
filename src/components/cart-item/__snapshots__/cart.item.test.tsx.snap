// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Item  renders a snapshot of <CartItem /> when it is passed into a component  1`] = `
<tr>
  <th
    scope="row"
  >
    1
  </th>
  <td
    className="product_image"
  >
    <img
      alt="product title"
      src="product image url"
    />
  </td>
  <td
    className="product_title"
  >
    product title
  </td>
  <td
    className="product_price"
  >
    $
    10
  </td>
  <td
    onClick={[Function]}
  >
    <p
      className="btn btn-danger"
    >
      Remove
    </p>
  </td>
</tr>
`;
